AK/LogStream.h

```diff
@@ -57,17 +58,66 @@ class LogStream {
 #endif
 };
 
-class DebugLogStream final : public LogStream {
+class BufferedLogStream : public LogStream {
+    mutable int m_size{0};
+    mutable int m_capacity{128};
+    union
+    {
+        mutable u8* m_buffer{nullptr};
+        mutable u8 m_local_buffer[128];
+    } u;
+
+    void grow(int byte_needed) const
```

Comment:
s/byte_needed/bytes_needed/ (or perhaps "capacity")
Also, this seems like it should be a `size_t` instead of an `int`.

Reply:
I initially had it as `size_t` but ended up changing it to `int` because the `write` function takes an `int`, so I ended up with lots of casting.

Reply:
Yeah you're right, we can change it. I think it had a slightly different implementation before where it being `int` was easier. I'll change it.